e2e/explorer/api/src/app/etherscan/contract.ts

```diff
@@ -38,11 +39,27 @@ async function verifySourceCode(params: GetParams<VerifySourceCode>) {
 
 export function getAbi({ address }: GetParams<GetABI>) {
   if (!address) throw new Error('Invalid Address format');
+  const account = store.addresses[address];
+  if(!isContract(account)) throw new Error('Contract source code not verified');
+
+  return store.artifacts[account.artifact].abi;
+}
 
+export function getSourceCode({ address }: GetParams<GetSourceCode>) {
+  if (!address) throw new Error('Invalid Address format');
   const account = store.addresses[address];
-  if (isContract(account)) {
-    return store.artifacts[account.artifact].abi;
-  } else {
-    throw new Error('Contract source code not verified');
-  }
+  if (!isContract(account)) throw new Error('Contract source code not verified');
+  
+  let optimization: "0" | "1"
+  if (store.builds[account.artifact].optimizationUsed === "false") optimization = "0"
```

Comment:
`store.builds[account.artifact].optimizationUsed` is a boolean not a string. You can just do : 
```suggestion
  const optimization = store.builds[account.artifact].optimizationUsed ? '1' : '0';
```